azmq/detail/config/thread.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
+
+    This file is part of azmq
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
+#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
+
+#if !defined(AZMQ_HAS_STD_THREAD)
```

Comment:
It might be more accurate to say, DISABLE_XXX macros which use Boost to fix
issues with broken standard library features would not be supported in the
AZMQ_USE_STANDALONE_ASIO scenario. The reasoning being that, if the goal of
using stand alone Asio is to have no dependencies beyond Asio and the
standard library, then you are actively asserting that you don't want to
rely on  Boost at all.

Those interested in standalone Asio can correct me if this stance is too
draconian, but it does seem to simplify the possible set of configurations
for which there need to be tests/builds/etc.


On Fri, Apr 3, 2015 at 5:17 AM, Andrey Upadyshev <notifications@github.com>
wrote:

> In azmq/detail/config/thread.hpp
> <https://github.com/zeromq/azmq/pull/71#discussion_r27723528>:
>
> > @@ -0,0 +1,26 @@
> > +/*
> > +    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
> > +
> > +    This file is part of azmq
> > +
> > +    Distributed under the Boost Software License, Version 1.0. (See accompanying
> > +    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
> > +*/
> > +#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
> > +#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
> > +
> > +#if !defined(AZMQ_HAS_STD_THREAD)
>
> I agree with AZMQ_USE_STANDALONE_ASIO.
>
> About completely boost free azmq. Do you mean that DISABLE_XXX macro will
> not supported for this case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/azmq/pull/71/files#r27723528>.
>